tests/performance/reindexing/reindexing_and_feeding.rb

```diff
@@ -0,0 +1,254 @@
+  def get_reindexing_status
+    status = vespa.clustercontrollers["0"].get_reindexing_json
+    return nil if status.nil?
+    return status['status'].first
+  end
+
+
+  def get_graphs
+    @test_config.map do |config, metrics|
+      metrics.map do |metric, limits|
+        {
+          :x => "blank",
+          :y => metric,
+          :title => " #{config[:legend]} #{metric}",
+          :filter => { "legend" => config[:legend] },
+          :historic => true
+        }.merge(limits)
+      end
+    end.flatten
+  end
+
+  # Wait for convergence of all services in the application — specifically document processors 
+  def wait_for_convergence(generation)
+    start_time = Time.now
+    until get_json(http_request(URI(application_url + "serviceconverge"), {}))["converged"] or Time.now - start_time > 60 # seconds
+      sleep 1
+    end
+    assert(Time.now - start_time < 60, "Services should converge on new generation within the minute")
+    assert(generation == get_json(http_request(URI(application_url + "serviceconverge"), {}))["wantedGeneration"],
+           "Should converge on generation #{generation}")
+    puts "Services converged on new config generation after #{Time.now - start_time} seconds"
+  end
+
+  # Trigger reindexing of the whole corpus
+  def trigger_reindexing
+    # Read baseline reindexing status — very first reindexing is a no-op in the reindexer controller
+    response = http_request(URI(application_url + "reindexing"), {})
+    assert(response.code.to_i == 200, "Request should be successful")
+    previous_reindexing_timestamp = get_json(response)["status"]["readyMillis"]
+
+    # Allow the reindexing maintainer some time to run, and mark the first no-op reindexing as done
+    sleep 60
+
+    # Trigger reindexing through reindexing API in /application/v2, and verify it was triggered
+    response = http_request_post(URI(application_url + "reindex"), {})
+    assert(response.code.to_i == 200, "Request should be successful")
+
+    response = http_request(URI(application_url + "reindexing"), {})
+    assert(response.code.to_i == 200, "Request should be successful")
+    current_reindexing_timestamp = get_json(response)["status"]["readyMillis"]
+    assert(previous_reindexing_timestamp < current_reindexing_timestamp,
+           "Previous reindexing timestamp (#{previous_reindexing_timestamp}) should be after current (#{current_reindexing_timestamp})")
+
+    deploy_app(@app)
+    wait_for_reindexing_start(current_reindexing_timestamp)
+  end
+
+  # Application and tenant names change based on the context this is run in.
+  def application_url
+    tenant = use_shared_configservers ? @tenant_name : "default"
+    application = use_shared_configservers ? @application_name : "default"
+    "https://#{vespa.nodeproxies.first[1].addr_configserver[0]}:#{19071}/application/v2/tenant/#{tenant}/application/#{application}/environment/prod/region/default/instance/default/"
```

Comment:
Set scheme to `http` to allow test to run without Vespa mTLS (underlying http(s) client will use correct protocol based on Vespa TLS settings).

Reply:
OK.